src/Tools/dotnet-counters/Program.cs

```diff
@@ -145,6 +155,19 @@ private static Task<int> Main(string[] args)
                 .AddCommand(ProcessStatusCommandHandler.ProcessStatusCommand("Lists the dotnet processes that can be monitored"))
                 .UseDefaults()
                 .Build();
+
+            ParseResult parseResult = parser.Parse(args);
+            string parsedCommandName = parseResult.CommandResult.Command.Name;
+            if (parsedCommandName == "monitor" || parsedCommandName == "collect")
+            {
+                IReadOnlyCollection<string> unparsedTokens = parseResult.UnparsedTokens;
```

Comment:
We've run into issues before where the transit from shell to S.CommandLine to usage has introduced subtle formatting issues for CLI args.  For example, we had those issues with escape character differences between bash, pwsh, and zsh that led to quotes being dropped or preserved.

Do we have a sense of whether we might be causing these sorts of issues for the child process?  For example, if I invoke the following:
```
dotnet counters monitor -- ./myapp --arg1 "The quick brown fox jumps over the lazy dog" --arg 2
```

will the child process be invoked as
```
./myapp --arg1 The quick brown fox jumps over the lazy dog --arg 2
```
or
```
./myapp --arg1 "The quick brown fox jumps over the lazy dog" --arg 2
```

Reply:
Thanks - the latest commit to this should address it.

Reply:
Can you double check that something complicated like 
```
dotnet trace collect -- ./myapp --arg "this ^-- ===is \t \r\n something;;; \r\n \n complicated"
```
works?  A pathologic test might be to try running dotnet-trace sa the subprocess and have a complex filter string on DiagnosticSourceEventSource that contains `\r\n`, `;`, and `"` in it.

Reply:
I have a local test that validates args: 
```csharp
using System;
using System.IO;

namespace argvalid
{
    class Program
    {
        static void Main(string[] args)
        {
            using (StreamWriter file = new StreamWriter(@"E:\result.txt")) {
                for (int i = 0; i < args.Length; i++) file.WriteLine($"args {i}: {args[i]}");
            }
        }
    }
}
```

Output from `argvalid.exe "this ^-- ===is \t \r\n something;;; \r\n \n complicated":
```
args 0: this ^-- ===is \t \r\n something;;; \r\n \n complicated
```

Output from `E:\diagnostics>dotnet-trace collect -- \temp\argvalid\bin\Debug\net5.0\argvalid.exe "this ^-- ===is \t \r\n something;;; \r\n \n complicated"`:

```
args 0: this ^-- ===is \t \r\n something;;; \r\n \n complicated
```

Reply:
Awesome!